docs/ReverseProxyToTor.md

```diff
@@ -215,6 +215,135 @@ systemctl restart nginx
 
 Now, visiting `mydomain.com` should show your BTCPay Server instance.
 
+
+## Do all this in a Docker container
+
+Ready made docker image 
+
+  https://hub.docker.com/r/cloudgenius/socator
+
+Code behind the image
+
+  https://github.com/beacloudgenius/socator
+
+
+### SocaTor = SOCAT + TOR
+Based on https://github.com/Arno0x/Docker-Socator
+
+It uses socat to listen on a given TCP port (5000 in this example) and to redirect incoming traffic to a Tor hidden service specified through environment variables. It acts as a relay between the standard web and a hidden service on the Tor network. You can optionally restrict the IP addresses that are allowed to connect to this service by specifying an `ALLOWED_RANGE` environment variable and using CIDR notation.
+
+### Usage
+
+Break free from cloud services providers limitations, secure and protect your bitcoin full node, connect that with a BTC Pay server, all behind TOR.
+Selectively expose the BTC Pay Server payment gateway and API to clearnet using socat+tor running on the Internet.
+
+--------------
+
+##### build
+
+```sh
+docker build -t cloudgenius/socator .
+```
+
+##### push
+
+```sh
+docker push cloudgenius/socator
+``
+
+##### Start the image in background (*daemon mode*) with IP address restriction:
+
+```sh
+docker run -d \
+        -p 5000:5000 \
+        -e "ALLOWED_RANGE=192.168.1.0/24" \
+        -e "TOR_SITE=zqktlwiuavvvqqt4ybvgvi7tyo4hjl5xgfuvpdf6otjiycgwqbym2qad.onion" \
+        -e "TOR_SITE_PORT=80" \
+        --name socator \
+        cloudgenius/socator
+```
+
+##### Start the image in foreground:
+
+```sh
+docker run --rm -ti \
+        -p 5000:5000 \
+        -e "TOR_SITE=zqktlwiuavvvqqt4ybvgvi7tyo4hjl5xgfuvpdf6otjiycgwqbym2qad.onion" \
+        -e "TOR_SITE_PORT=80" \
+        --name socator \
+        cloudgenius/socato
+```
+
+Now http://localhost:5000 should show you the tor hidden service you specified in the above command.
+
+
+
+
+## Use that Docker container in a Kubernetes Cluster using these manifests
+
+```yaml
+---
```

Comment:
remove dividers and make each one separate yaml code block

Reply:
Manifests separated.

Reply:
just suggesting this to make it more beginner friendly :)